File: Frontend/angular-app/src/app/modules/admin/repositories/account/commands/update-account-command.ts
import {Inject} from "@angular/core";
import {Account} from "../../../entities/account";
import {ServiceResult} from "../../../../../core/models/service-result";
import {IRequest, IRequestHandler} from "../../../../../core/services/mediator/interfaces";
import {HttpService} from "../../../../../core/services/http/http.service";
import {ApplicationError} from "../../../../../core/exceptions/application-error";
import {MediatorHandler} from "../../../../../core/services/mediator/decorator";
import {ValidationRule} from "../../../../../core/validation/validation-rule";
import {HttpRequest} from "../../../../../core/services/http/http-request";
import {UpdatePersonCommand} from "../../person/commands/update-person-command";

export class UpdateAccountCommand extends IRequest<UpdateAccountCommand, Account> {
  public id: number | undefined = undefined;
  public title: string | undefined = undefined;
  public code: string | undefined = undefined;
  public type: number | undefined = undefined;
  public isActive: boolean | undefined = undefined;
  public person: UpdatePersonCommand = new UpdatePersonCommand();
  constructor() {
    super();
  }

  mapFrom(entity: Account): UpdateAccountCommand {
    throw new ApplicationError(UpdateAccountCommand.name, this.mapTo.name, 'Not Implemented Yet')
  }

  mapTo(): Account {
    throw new ApplicationError(UpdateAccountCommand.name, this.mapTo.name, 'Not Implemented Yet')
  }

  get url(): string {
    return "/admin/accounts/update";
  }

  get validationRules(): ValidationRule[] {
    return [];
  }
}

@MediatorHandler(UpdateAccountCommandHandler.name)
export class UpdateAccountCommandHandler implements IRequestHandler<UpdateAccountCommand, Account> {
  constructor(
    @Inject(HttpService) private _httpService: HttpService,
  ) {
  }

  async Handle(request: UpdateAccountCommand): Promise<Account> {
    let httpRequest: HttpRequest<UpdateAccountCommand> = new HttpRequest<UpdateAccountCommand>(request.url, request);
    // httpRequest.Query  += ;
    // httpRequest.Headers = httpRequest.Headers.append()

    return await this._httpService.Put<UpdateAccountCommand, ServiceResult<Account>>(httpRequest).toPromise().then(response => {
      return response.objResult
    })
  }
}

File: Backend/Microservices/Admin/Eefa.Admin.Application/UseCases/Accounts/Commands/Update/UpdateAccountCommand.cs
using AutoMapper;
using Eefa.Admin.Application.CommandQueries.Accounts.Models;
using Eefa.Admin.Application.CommandQueries.Person.Command.Update;
using Eefa.Admin.Data.Databases.Entities;
using Library.Common;
using Library.ConfigurationAccessor.Mappings;
using Library.Interfaces;
using Library.Models;
using MediatR;
using Microsoft.EntityFrameworkCore;
using System.Threading;
using System.Threading.Tasks;

namespace Eefa.Admin.Application.CommandQueries.Accounts.Commands.Update
{
    public class UpdateAccountCommand : CommandBase, IRequest<ServiceResult>, IMapFrom<UpdateAccountCommand>, ICommand
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Code { get; set; }
        public int Type { get; set; }
        public bool IsActive { get; set; }

        public UpdatePersonCommand Person { get; set; }
        public void Mapping(Profile profile)
        {
            profile.CreateMap<UpdateAccountCommand, Data.Databases.Entities.AccountReference>()
                .IgnoreAllNonExisting();
        }
    }

    public class UpdateAccountCommandHandler : IRequestHandler<UpdateAccountCommand, ServiceResult>
    {
        private readonly IRepository _repository;
        private readonly IMapper _mapper;
        private readonly IMediator _mediator;

        public UpdateAccountCommandHandler(IRepository repository, IMapper mapper, IMediator mediator)
        {
            _mapper = mapper;
            _mediator = mediator;
            _repository = repository;
        }

        public async Task<ServiceResult> Handle(UpdateAccountCommand request, CancellationToken cancellationToken)
        {
            var account = await _repository
                 .Find<AccountReference>(c =>
             c.ObjectId(request.Id))
                 .Include(x => x.Person)
             .FirstOrDefaultAsync(cancellationToken);

            account.Code = request.Code;
            account.Title = request.Title;
            account.IsActive = request.IsActive;

            if (account.Person == null)
            {

                var newPerson = _mapper.Map<Data.Databases.Entities.Person>(request.Person);

            }
            else
            {

            }

            await _repository.SaveChangesAsync();

            return ServiceResult.Success(_mapper.Map<AccountModel>(account));
        }
    }
}
    
